src/state/governance/hooks.ts

```diff
@@ -440,6 +443,68 @@ export function useVoteCallback(): {
   return { voteCallback }
 }
 
+export function useQueueCallback(): {
+  queueCallback: (proposalId: string | undefined) => undefined | Promise<string>
+} {
+  const { account, chainId } = useActiveWeb3React()
+
+  const latestGovernanceContract = useLatestGovernanceContract()
+
+  const addTransaction = useTransactionAdder()
+
+  const queueCallback = useCallback(
+    (proposalId: string | undefined) => {
+      if (!account || !latestGovernanceContract || !proposalId || !chainId) return
+      const args = [proposalId]
+      return latestGovernanceContract.estimateGas.queue(...args, {}).then((estimatedGasLimit) => {
+        return latestGovernanceContract
+          .queue(...args, { value: null, gasLimit: calculateGasMargin(estimatedGasLimit) })
+          .then((response: TransactionResponse) => {
+            addTransaction(response, {
+              type: TransactionType.QUEUE,
+              governorAddress: latestGovernanceContract.address,
+              proposalId: parseInt(proposalId),
+            })
+            return response.hash
+          })
+      })
+    },
+    [account, addTransaction, latestGovernanceContract, chainId]
+  )
+  return { queueCallback }
+}
+
+export function useExecuteCallback(): {
```

Comment:
In keeping with other hooks here, this should return the callback, not the callback wrapped in an object.